// @videogular/ngx-videogular/esm2020/controls/lib/controls.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { VgControlsComponent } from './components/vg-controls/vg-controls.component';
import { VgVolumeComponent } from './components/vg-volume/vg-volume.component';
import { VgTrackSelectorComponent } from './components/vg-track-selector/vg-track-selector.component';
import { VgTimeDisplayComponent, VgUtcPipe } from './components/vg-time-display/vg-time-display.component';
import { VgScrubBarComponent } from './components/vg-scrub-bar/vg-scrub-bar.component';
import { VgQualitySelectorComponent } from './components/vg-quality-selector/vg-quality-selector.component';
import { VgPlaybackButtonComponent } from './components/vg-playback-button/vg-playback-button.component';
import { VgPlayPauseComponent } from './components/vg-play-pause/vg-play-pause.component';
import { VgMuteComponent } from './components/vg-mute/vg-mute.component';
import { VgFullscreenComponent } from './components/vg-fullscreen/vg-fullscreen.component';
import { VgCoreModule } from '@videogular/ngx-videogular/core';
import { VgScrubBarBufferingTimeComponent } from './components/vg-scrub-bar/vg-scrub-bar-buffering-time/vg-scrub-bar-buffering-time.component';
import { VgScrubBarCuePointsComponent } from './components/vg-scrub-bar/vg-scrub-bar-cue-points/vg-scrub-bar-cue-points.component';
import { VgScrubBarCurrentTimeComponent } from './components/vg-scrub-bar/vg-scrub-bar-current-time/vg-scrub-bar-current-time.component';
import * as i0 from "@angular/core";
const components = [
    VgControlsComponent,
    VgVolumeComponent,
    VgTrackSelectorComponent,
    VgTimeDisplayComponent,
    VgScrubBarComponent,
    VgQualitySelectorComponent,
    VgPlaybackButtonComponent,
    VgPlayPauseComponent,
    VgMuteComponent,
    VgFullscreenComponent,
    VgUtcPipe,
    VgScrubBarBufferingTimeComponent,
    VgScrubBarCuePointsComponent,
    VgScrubBarCurrentTimeComponent
];
export class VgControlsModule {
}
/** @nocollapse */ /** @nocollapse */ VgControlsModule.ɵfac = function VgControlsModule_Factory(t) { return new (t || VgControlsModule)(); };
/** @nocollapse */ /** @nocollapse */ VgControlsModule.ɵmod = /** @pureOrBreakMyCode */ i0.ɵɵdefineNgModule({ type: VgControlsModule });
/** @nocollapse */ /** @nocollapse */ VgControlsModule.ɵinj = /** @pureOrBreakMyCode */ i0.ɵɵdefineInjector({ imports: [[CommonModule, VgCoreModule]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(VgControlsModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, VgCoreModule],
                declarations: [...components],
                exports: [...components],
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(VgControlsModule, { declarations: [VgControlsComponent,
        VgVolumeComponent,
        VgTrackSelectorComponent,
        VgTimeDisplayComponent,
        VgScrubBarComponent,
        VgQualitySelectorComponent,
        VgPlaybackButtonComponent,
        VgPlayPauseComponent,
        VgMuteComponent,
        VgFullscreenComponent,
        VgUtcPipe,
        VgScrubBarBufferingTimeComponent,
        VgScrubBarCuePointsComponent,
        VgScrubBarCurrentTimeComponent], imports: [CommonModule, VgCoreModule], exports: [VgControlsComponent,
        VgVolumeComponent,
        VgTrackSelectorComponent,
        VgTimeDisplayComponent,
        VgScrubBarComponent,
        VgQualitySelectorComponent,
        VgPlaybackButtonComponent,
        VgPlayPauseComponent,
        VgMuteComponent,
        VgFullscreenComponent,
        VgUtcPipe,
        VgScrubBarBufferingTimeComponent,
        VgScrubBarCuePointsComponent,
        VgScrubBarCurrentTimeComponent] }); })();
