// @videogular/ngx-videogular/esm2020/modulo/lib/modulo.module.mjs
import { NgModule } from "@angular/core";
import { CommonModule } from "@angular/common";
import { VgModuloComponent } from "./components/vg-modulo.component";
import * as i0 from "@angular/core";
export class VgModuloModule {
}
/** @nocollapse */ /** @nocollapse */ VgModuloModule.ɵfac = function VgModuloModule_Factory(t) { return new (t || VgModuloModule)(); };
/** @nocollapse */ /** @nocollapse */ VgModuloModule.ɵmod = /** @pureOrBreakMyCode */ i0.ɵɵdefineNgModule({ type: VgModuloModule });
/** @nocollapse */ /** @nocollapse */ VgModuloModule.ɵinj = /** @pureOrBreakMyCode */ i0.ɵɵdefineInjector({ imports: [[
            CommonModule
        ]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(VgModuloModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [VgModuloComponent],
                exports: [VgModuloComponent],
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(VgModuloModule, { declarations: [VgModuloComponent], imports: [CommonModule], exports: [VgModuloComponent] }); })();
