// @videogular/ngx-videogular/esm2020/streaming/lib/streaming.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { VgDashDirective } from './directives/vg-dash/vg-dash.directive';
import { VgHlsDirective } from './directives/vg-hls/vg-hls.directive';
import { VgCoreModule } from '@videogular/ngx-videogular/core';
import * as i0 from "@angular/core";
export class VgStreamingModule {
}
/** @nocollapse */ /** @nocollapse */ VgStreamingModule.ɵfac = function VgStreamingModule_Factory(t) { return new (t || VgStreamingModule)(); };
/** @nocollapse */ /** @nocollapse */ VgStreamingModule.ɵmod = /** @pureOrBreakMyCode */ i0.ɵɵdefineNgModule({ type: VgStreamingModule });
/** @nocollapse */ /** @nocollapse */ VgStreamingModule.ɵinj = /** @pureOrBreakMyCode */ i0.ɵɵdefineInjector({ imports: [[CommonModule, VgCoreModule]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(VgStreamingModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, VgCoreModule],
                declarations: [VgDashDirective, VgHlsDirective],
                exports: [VgDashDirective, VgHlsDirective]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(VgStreamingModule, { declarations: [VgDashDirective, VgHlsDirective], imports: [CommonModule, VgCoreModule], exports: [VgDashDirective, VgHlsDirective] }); })();
