// @videogular/ngx-videogular/esm2020/core/lib/core.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { VgApiService } from './services/vg-api/vg-api.service';
import { VgControlsHiddenService } from './services/vg-controls-hidden/vg-controls-hidden.service';
import { VgFullscreenApiService } from './services/vg-fullscreen-api/vg-fullscreen-api.service';
import { VgUtilsService } from './services/vg-utils/vg-utils.service';
import { VgEvents } from './services/events/vg-events.service';
import { VgStates } from './services/states/vg-states.service';
import { VgCuePointsDirective } from './directives/vg-cue-points/vg-cue-points.directive';
import { VgMediaDirective } from './directives/vg-media/vg-media.directive';
import { VgPlayerComponent } from './components/vg-player/vg-player.component';
import * as i0 from "@angular/core";
const services = [
    VgApiService,
    VgControlsHiddenService,
    VgFullscreenApiService,
    VgUtilsService,
    VgEvents,
    VgStates
];
const directives = [
    VgCuePointsDirective,
    VgMediaDirective
];
export class VgCoreModule {
}
/** @nocollapse */ /** @nocollapse */ VgCoreModule.ɵfac = function VgCoreModule_Factory(t) { return new (t || VgCoreModule)(); };
/** @nocollapse */ /** @nocollapse */ VgCoreModule.ɵmod = /** @pureOrBreakMyCode */ i0.ɵɵdefineNgModule({ type: VgCoreModule });
/** @nocollapse */ /** @nocollapse */ VgCoreModule.ɵinj = /** @pureOrBreakMyCode */ i0.ɵɵdefineInjector({ providers: [...services], imports: [[CommonModule]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(VgCoreModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule],
                providers: [...services],
                declarations: [...directives, VgPlayerComponent],
                exports: [...directives, VgPlayerComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(VgCoreModule, { declarations: [VgCuePointsDirective,
        VgMediaDirective, VgPlayerComponent], imports: [CommonModule], exports: [VgCuePointsDirective,
        VgMediaDirective, VgPlayerComponent] }); })();
