// @videogular/ngx-videogular/esm2020/overlay-play/lib/overlay-play.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { VgOverlayPlayComponent } from './vg-overlay-play.component';
import { VgCoreModule } from '@videogular/ngx-videogular/core';
import * as i0 from "@angular/core";
export class VgOverlayPlayModule {
}
/** @nocollapse */ /** @nocollapse */ VgOverlayPlayModule.ɵfac = function VgOverlayPlayModule_Factory(t) { return new (t || VgOverlayPlayModule)(); };
/** @nocollapse */ /** @nocollapse */ VgOverlayPlayModule.ɵmod = /** @pureOrBreakMyCode */ i0.ɵɵdefineNgModule({ type: VgOverlayPlayModule });
/** @nocollapse */ /** @nocollapse */ VgOverlayPlayModule.ɵinj = /** @pureOrBreakMyCode */ i0.ɵɵdefineInjector({ imports: [[CommonModule, VgCoreModule]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(VgOverlayPlayModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, VgCoreModule],
                declarations: [VgOverlayPlayComponent],
                exports: [VgOverlayPlayComponent],
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(VgOverlayPlayModule, { declarations: [VgOverlayPlayComponent], imports: [CommonModule, VgCoreModule], exports: [VgOverlayPlayComponent] }); })();
