// @videogular/ngx-videogular/esm2020/ima-ads/lib/ima-ads.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { VgImaAdsComponent } from './vg-ima-ads.component';
import { VgCoreModule } from '@videogular/ngx-videogular/core';
import * as i0 from "@angular/core";
export class VgImaAdsModule {
}
/** @nocollapse */ /** @nocollapse */ VgImaAdsModule.ɵfac = function VgImaAdsModule_Factory(t) { return new (t || VgImaAdsModule)(); };
/** @nocollapse */ /** @nocollapse */ VgImaAdsModule.ɵmod = /** @pureOrBreakMyCode */ i0.ɵɵdefineNgModule({ type: VgImaAdsModule });
/** @nocollapse */ /** @nocollapse */ VgImaAdsModule.ɵinj = /** @pureOrBreakMyCode */ i0.ɵɵdefineInjector({ imports: [[CommonModule, VgCoreModule]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(VgImaAdsModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, VgCoreModule],
                declarations: [VgImaAdsComponent],
                exports: [VgImaAdsComponent],
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(VgImaAdsModule, { declarations: [VgImaAdsComponent], imports: [CommonModule, VgCoreModule], exports: [VgImaAdsComponent] }); })();
